kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt

```diff
@@ -1,18 +1,277 @@
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.kwarchart.android.Chart
+import com.kwarchart.android.ChartCanvas
+import com.kwarchart.android.enum.LegendPosition
+import com.kwarchart.android.enum.LineChartType
+import com.kwarchart.android.model.BarSeries
+import com.kwarchart.android.model.ChartData
+import com.kwarchart.android.model.Legend
+import com.kwarchart.android.model.LineSeries
+import com.kwarchart.android.util.PathUtils
+import android.graphics.PorterDuff
+
+import android.graphics.PorterDuffColorFilter
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.graphics.painter.ColorPainter
+
+import com.kwarchart.android.*
+import com.kwarchart.android.*
+
+
+private var mHGap = 0f
+private var mVGap = 0f
+private var mMaxVal = 0f
+private var mMaxLen = 0
 
 /**
  *
  */
 @Composable
-fun BarChart() {
+fun <T> BarChart(
+        modifier: Modifier = Modifier,
+        data: List<BarSeries<T>>,
+        xAxisName: String? = null,
+        yAxisName: String? = null,
+        axesColor: Color = Color.Gray,
+        showAxes: Boolean = true,
+        gridsColor: Color = Color.Gray,
+        showGrid: Boolean = true,
+        legendPos: LegendPosition? = null
+) {
+    val keys = mutableListOf<T>()
+
+    data.forEach {
+        if (mMaxLen < it.data.size) {
+            mMaxLen = it.data.size
+        }
+
+        it.data.forEach { chartData ->
+            if (mMaxVal < chartData.value) {
+                mMaxVal = chartData.value
+            }
+
+            if (!keys.contains(chartData.key)) {
+                keys.add(chartData.key)
+            }
+        }
+    }
+
+    Chart(
+            modifier = modifier,
+            xAxisName = xAxisName,
+            yAxisName = yAxisName,
+            legend = legendPos?.let { _ ->
+                val tmpLegend = Legend(legendPos)
+                data.forEach {
+                    tmpLegend.legends.add(it.legend)
+                    tmpLegend.colors.add(it.color)
+                }
+                tmpLegend
+            }
+    ) {
+        Canvas(
+                modifier = modifier
+                        .weight(5f)
+                        .padding(
+                                start = (mMaxVal.toString().length * 8).dp,
+                                top = 10.dp,
+                                end = 10.dp,
```

Comment:
create a diff function for this

Reply:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.